### PR TITLE
refactor: consolidate and streamline unit test infrastructure

### PR DESCRIPTION
**Pull request by SKannaniOS**
## Description
This PR consolidates the unit test infrastructure by removing duplicate mock provider classes and streamlining the testing utilities. The changes primarily focus on refactoring the test helpers to use a unified approach, removing redundant code, and improving test maintainability across the SDK.

The main changes include:
- **Merged duplicate mock providers**: Consolidated `SwiftTestMockProvider` functionality into the existing `MockProvider` class
- **Removed redundant test utilities**: Eliminated duplicate helper functions and mock implementations
- **Standardized mock usage**: Updated all test files to use the consolidated `MockProvider` instead of the Swift Testing-specific variant
- **Added new mock plugin**: Introduced `MockEventCapturePlugin` for better event testing capabilities
- **Improved test organization**: Cleaned up test structure and removed unnecessary `@MainActor` annotations where not needed

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [x] Code refactor/optimization

## Implementation Details
- **Consolidated MockProvider**: Merged `SwiftTestMockProvider.swift` into the existing `MockProvider.swift`, eliminating code duplication and providing a single source of truth for test mocks
- **Enhanced MockProvider capabilities**: Added comprehensive mock event generation, configuration creation, and URL session management utilities
- **Introduced MockEventCapturePlugin**: Created a new robust mock plugin for capturing and validating events during testing with advanced filtering and modification capabilities
- **Updated MockLogger**: Moved `SwiftMockLogger` functionality into `MockProvider.swift` as `MockLogger` for consistency
- **Removed @MainActor annotations**: Cleaned up test classes by removing unnecessary `@MainActor` annotations that were causing test execution issues
- **Standardized test imports**: Updated all test files to import `Testing` instead of `XCTest` where applicable
- **Added watchOS compatibility**: Added conditional compilation flags to skip URLProtocol-based tests on watchOS where they don't work
- **Simplified test initialization**: Reduced boilerplate code in test initializers by leveraging the consolidated mock provider
- **Enhanced error handling**: Improved mock implementations to better handle edge cases and error scenarios

## Checklist
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation (if appropriate).
- [x] I have ensured that my code follows the project's code style.
- [x] I have checked for potential performance impacts and optimized if necessary.
- [x] I have checked the code for security issues.
- [ ] I have updated the changelog (if required).

## How to test?
1. **Run the full test suite**: Execute all unit tests to ensure no regressions were introduced
   ```bash
   xcodebuild test -workspace RudderStackAnalytics.xcworkspace -scheme RudderStackAnalytics -destination 'platform=iOS Simulator,name=iPhone 15'
   ```

2. **Verify mock functionality**: Run specific test files that use the consolidated MockProvider:
   ```bash
   # Test event processing with new mocks
   xcodebuild test -workspace RudderStackAnalytics.xcworkspace -scheme RudderStackAnalytics -destination 'platform=iOS Simulator,name=iPhone 15' -only-testing:RudderStackAnalyticsTests/EventQueueTests
   
   # Test plugin functionality
   xcodebuild test -workspace RudderStackAnalytics.xcworkspace -scheme RudderStackAnalytics -destination 'platform=iOS Simulator,name=iPhone 15' -only-testing:RudderStackAnalyticsTests/AppInfoPluginTests
   ```

3. **Cross-platform testing**: Verify that tests run correctly on all supported platforms:
   ```bash
   # iOS
   xcodebuild test -workspace RudderStackAnalytics.xcworkspace -scheme RudderStackAnalytics -destination 'platform=iOS Simulator,name=iPhone 15'
   
   # macOS
   xcodebuild test -workspace RudderStackAnalytics.xcworkspace -scheme RudderStackAnalytics -destination 'platform=macOS'
   
   # tvOS
   xcodebuild test -workspace RudderStackAnalytics.xcworkspace -scheme RudderStackAnalytics -destination 'platform=tvOS Simulator,name=Apple TV'
   
   # watchOS
   xcodebuild test -workspace RudderStackAnalytics.xcworkspace -scheme RudderStackAnalytics -destination 'platform=watchOS Simulator,name=Apple Watch Series 9 (45mm)'
   ```

4. **Validate new MockEventCapturePlugin**: Test the event capture capabilities:
   - Verify event filtering works correctly
   - Test event modification functionality
   - Ensure thread-safe event capturing
   - Validate async event waiting mechanisms

## Breaking Changes
This is a **non-breaking change** for the public API. All changes are internal to the test infrastructure and do not affect the public SDK interfaces or functionality.

**For SDK contributors/maintainers:**
- Test files now use `MockProvider` instead of `SwiftTestMockProvider`
- Some test helper utilities have been relocated within the `MockProvider` class
- The new `MockEventCapturePlugin` should be used for event-related testing going forward

## Maintainers Checklist
- [ ] The code has been reviewed.
- [ ] CI tests have passed.
- [ ] All necessary documentation has been updated.

## Additional Context
This refactoring addresses several pain points in the test infrastructure:

1. **Code duplication**: Previously, we had two similar mock provider classes (`MockProvider` and `SwiftTestMockProvider`) that provided overlapping functionality
2. **Inconsistent testing patterns**: Different test files used different mock utilities, making maintenance difficult
3. **Platform compatibility**: Some tests failed on watchOS due to URLProtocol limitations
4. **Test execution issues**: Unnecessary `@MainActor` annotations caused concurrency-related test failures

The consolidated approach provides:
- **Single source of truth** for all test mocks and utilities
- **Better platform support** with conditional compilation
- **Enhanced debugging capabilities** through improved mock plugins
- **Cleaner test code** with reduced boilerplate

This refactoring sets the foundation for more maintainable and reliable unit tests going forward, while ensuring no functionality is lost in the process.

### CODE REVIEW

File: Tests/RudderStackAnalyticsTests/Base/AnalyticsTests.swift [hunk 1]
```diff
@@ -368,16 +367,12 @@ class AnalyticsTests {
...
-            
-            self.analytics.flush()
-            await runAfter(0.3) {
-                #expect(self.mockStorage.batchCount == 0)
-            }
-        }
+        analytics.flush()
+        
+        #expect(analytics.isFlushed)
     }
```
[the remark this answers is not in the document]
Comment (SKannaniOS) [reply Tests/RudderStackAnalyticsTests/Base/AnalyticsTests.swift]: Updated

File: Tests/RudderStackAnalyticsTests/Helpers/Base/MockEventCapturePlugin.swift [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+//  RudderStackAnalyticsTests
+//
+//  Created by Satheesh Kannan on 13/11/25.
+//
+
+import Foundation
+@testable import RudderStackAnalytics
+
+// MARK: - MockEventCapturePlugin
+final class MockEventCapturePlugin: Plugin {
```
Comment (1abhishekpandey) [inline Tests/RudderStackAnalyticsTests/Helpers/Base/MockEventCapturePlugin.swift:12]: Are we using this class anywhere, as of now? If yes, just point me to that code.

File: Tests/RudderStackAnalyticsTests/Plugins/DeviceMode/CustomIntegrationPluginTests.swift [hunk 3]
```diff
@@ -10,11 +10,19 @@ import Testing
...
 struct CustomIntegrationPluginTests {
     
+    var analytics: Analytics
+    
+    init() {
+        let mockConfiguration = MockProvider.createMockConfiguration()
+        mockConfiguration.flushPolicies = []
+        
+        self.analytics = Analytics(configuration: mockConfiguration)
+    }
```
Comment (1abhishekpandey) [inline Tests/RudderStackAnalyticsTests/Plugins/DeviceMode/CustomIntegrationPluginTests.swift:20]: Since the nature of this change is different, it could have been made in a new PR to reduce the no. of files changed. Please consider that in future.

File: Tests/RudderStackAnalyticsTests/Helpers/Base/MockEventCapturePlugin.swift (hunk 2 above)
Comment (SKannaniOS) [reply Tests/RudderStackAnalyticsTests/Helpers/Base/MockEventCapturePlugin.swift:12]: Please check `AnalyticsTests`.

**Review by 1abhishekpandey — APPROVED** [review]

**Comment by 1abhishekpandey** [comment]
@SKannaniOS There's one 1 new SonarQube issue. Can you please look into it?

**Comment by SKannaniOS** [comment]
> @SKannaniOS There's one 1 new SonarQube issue. Can you please look into it?

A separate PR will handle all SonarQube issues. I’ll address them together there.
